contrib/cirrus/runner.sh

```diff
@@ -299,6 +292,19 @@ dotest() {
                 -o CheckHostIP=no $GOSRC/$SCRIPT_BASE/runner.sh
         # does not return
     fi
+}
+
+# Handle local|remote integration|system testing in a uniform way
+dotest() {
+    local testsuite="$1"
+    req_env_vars testsuite CONTAINER TEST_ENVIRON PRIV_NAME
+
+    # shellcheck disable=SC2154
+    if ((CONTAINER==0)) && [[ "$TEST_ENVIRON" == "container" ]]; then
+        exec_container  # does not return
+    fi;
+
+    reexec_if_rootless
```

Comment:
Issue in the original logic (not new): I don't ever see us running in a container first, then going rootless via ssh.  More than likely it would be the other way around - we become rootless first, then run tests in a container.